custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,421 @@
+from subliminal_patch.providers import Provider
+from subliminal_patch.subtitle import Subtitle
+from subliminal_patch.exceptions import TooManyRequests
+from subliminal_patch.providers.utils import get_subtitle_from_archive
+from functools import cache
+from subzero.language import Language
+
+logger = logging.getLogger(__name__)
+
+class JimakuSubtitle(Subtitle):
+    '''Jimaku Subtitle.'''
+    provider_name = 'jimaku'
+    
+    hash_verifiable = False
+
+    def __init__(self, video, subtitle_id, subtitle_url, release_info, anilist_id=None):
+        # Override param 'language' as it could only ever be "ja"
+        super(JimakuSubtitle, self).__init__(Language("jpn"))
+        
+        self.video = video
+        self.subtitle_id = subtitle_id
+        self.subtitle_url = subtitle_url
+        self.release_info = release_info
+        self.anilist_id = anilist_id
+        
+    @property
+    def id(self):
+        return self.subtitle_id
+
+    def get_matches(self, video):
+        matches = set()
+        
+        if isinstance(video, Episode):
+            # series name
+            if sanitize(video.series) and sanitize(self.video.series) in (
+                    sanitize(name) for name in [video.series] + video.alternative_series):
+                matches.add('series')
+            
+            # season
+            if video.season and self.video.season is None or video.season and video.season == self.video.season:
+                matches.add('season')            
+
+            # year
+            if video.original_series and self.video.year is None or video.year and video.year == self.video.year:
+                matches.add('year')
+                
+            # episode
+            if video.episode == self.video.episode:
+                matches.add('episode')
+        else:
+            raise ValueError(f"Unhandled instance of argument 'video': {type(video)}")
+
+        return matches
+
+class JimakuProvider(Provider):
+    '''Jimaku Provider.'''
+    video_types = (Episode, Movie)
+    api_url = 'https://jimaku.cc/api'
+    
+    api_ratelimit_max_delay = 5 # In seconds
+    api_ratelimit_backoff_limit = 3
+    
+    # See _get_tvdb_anidb_mapping()
+    video_episode_number_override = None
+    
+    languages = {Language.fromietf("ja")}
+
+    def __init__(self, enable_archives, enable_ai_subs, api_key):
+        if api_key:
+            self.api_key = api_key
+        else:
+            raise ConfigurationError('Missing api_key.')
+
+        self.enable_archives = enable_archives
+        self.enable_ai_subs = enable_ai_subs
+        self.session = None
+
+    def initialize(self):
+        self.session = Session()
+        self.session.headers['User-Agent'] = 'Subliminal/%s' % __short_version__
+        self.session.headers['Content-Type'] = 'application/json'
+        self.session.headers['Authorization'] = self.api_key
+
+    def terminate(self):
+        self.session.close()
+
+    def query(self, video):
```

Comment:
I noticed that the other "private" methods you prefixed with `_` but not this one?

Reply:
Fixed in [`db5f56c`](https://github.com/morpheus65535/bazarr/pull/2505/commits/db5f56c2ffde69f2bde023fd79bbbdb0df575a20).